pcdsdevices/slits.py

```diff
@@ -0,0 +1,321 @@
+        Whether the slits are inserted into the beampath
+        """
+        return all([self.nominal_aperture[idx] > self.current_aperture[idx]
+                    for idx in range(0, 2)])
+
+    @property
+    def removed(self):
+        """
+        Whether the slits are entirely removed from the beampath
+        """
+        return not self.inserted
+
+    @property
+    def transmission(self):
+        """
+        Estimated transmission of the slits based on :attr:`.nominal_aperture`
+        """
+        # Find most restrictive side of slit aperture
+        min_dim = np.argmin(self.current_aperture)
+        # Don't allow transmissions over 1.0
+        return min([self.current_aperture[min_dim]
+                    / self.nominal_aperture[min_dim],
+                    1.0])
+
+    @property
+    def current_aperture(self):
+        """
+        Current size of the aperture (width, height)
+        """
+        return (self.xwidth.position, self.ywidth.position)
+
+    def remove(self, size=None, wait=False, timeout=None, **kwargs):
+        """
+        Open the slits to unblock the beam
+
+        Parameters
+        ----------
+        size : float, optional
+            Open the slits to a specific size otherwise
+            `:attr:`.nominal_aperture is used
+
+        wait : bool, optional
+            Wait for the status object to complete the move before returning
+
+        timeout : float, optional
+            Maximum time to wait for the motion. If None, the default timeout
+            for this positioner is used
+
+        Returns
+        -------
+        MoveStatus:
+            Status object based on move completion
+
+        See Also
+        --------
+        :meth:`Slits.move`
+        """
+        # Use nominal_aperture by default
+        size = size or self.nominal_aperture
+        return self.move(size, wait=wait, timeout=timeout, **kwargs)
+
+    def set(self, size):
+        """
+        Alias for the move method, here for bluesky compatibilty
+        """
+        return self.move(size, wait=False)
+
+    @property
+    def hints(self):
+        """Device hints"""
+        return {'fields': [self.xwidth.readback.name,
+                           self.ywidth.readback.name]}
+
+    def open(self):
+        """
+        Uses the built-in `OPEN` record to move open the aperture
+        """
+        self.open_cmd.put(1)
+
+    def close(self):
+        """
+        Close the slits to have an aperture of 0mm on each side
+        """
+        self.close_cmd.put(1)
+
+    def block(self):
+        """
+        Overlap the slits to block the beam
+        """
+        self.block_cmd.put(1)
+
+    def stage(self):
+        """
+        Store the initial values of the aperture position before scanning
+        """
+        self._original_vals[self.xwidth.setpoint] = self.xwidth.readback.value
+        self._original_vals[self.ywidth.setpoint] = self.ywidth.readback.value
+        return super().stage()
```

Comment:
Is this a workaround for not being able to declare `stage_sigs` without setting a value?

Reply:
Yeah that is the reason.